src/elements/content-explorer/ContentExplorer.js

```diff
@@ -292,6 +297,15 @@ class ContentExplorer extends Component<Props, State> {
         }
     }
 
+    componentDidUpdate() {
+        const { features }: Props = this.props;
+        const { viewMode }: State = this.state;
+
+        if (isFeatureEnabled(features, 'contentExplorer.gridView.enabled')) {
```

Comment:
Probably want to check if the new `viewMode` is different from `prevState.viewMode`.